### PR TITLE
Update expected safeties in Vecgeom tests and CMake version range

### PR DESCRIPTION
**Pull request by mrguilima**
Update CMake version range to 3.31.
Update expected values of safeties of a few Vecgeom test instances.

### CODE REVIEW

**Comment by sethrj** [comment]
Thanks for this @mrguilima ! It's going to have to wait until I update our CI, but I'm going to try to make headway on that during some downtime today...

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,8 @@ set(CMAKE_USER_MAKE_RULES_OVERRIDE
   "${CMAKE_CURRENT_LIST_DIR}/cmake/CeleritasMakeRulesOverride.cmake")
 
 project(Celeritas VERSION "${Celeritas_VERSION}" LANGUAGES CXX)
-cmake_policy(VERSION 3.18...3.28)
+cmake_policy(VERSION 3.18...3.31)
+cmake_policy(SET CMP0175 OLD)
```
Comment (sethrj) [inline CMakeLists.txt]: Please revert this line; instead, we should fix whatever is broken in the CMake policy. https://cmake.org/cmake/help/latest/policy/CMP0175.html#cmp0175

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,7 +14,8 @@ set(CMAKE_USER_MAKE_RULES_OVERRIDE
   "${CMAKE_CURRENT_LIST_DIR}/cmake/CeleritasMakeRulesOverride.cmake")
 
 project(Celeritas VERSION "${Celeritas_VERSION}" LANGUAGES CXX)
-cmake_policy(VERSION 3.18...3.28)
+cmake_policy(VERSION 3.18...3.31)
```
Comment (sethrj) [inline CMakeLists.txt]: It seems like this is redundant with the `cmake_minimum_required` above... maybe delete the cmake policy here? (I somehow thought it had to be after `project`)

**Comment by sethrj** [comment]
@mrguilima I think once you fix the remaining tests this should be good to go! https://github.com/celeritas-project/celeritas/actions/runs/12635001598/job/35204556112

File: CMakeLists.txt (hunk 2 above)
Comment (mrguilima) [reply CMakeLists.txt]: The build succeeds with CMP0175 set to OLD, but fails with NEW.  I will try to fix it...

Comment (sethrj) [reply CMakeLists.txt]: @mrguilima Don't worry about it. Can you build without these changes? It would be better for you to revert this file and  make this PR *just* about the VG update.

Comment (sethrj) [reply CMakeLists.txt]: Moved changes to #1563 .

File: test/geocel/vg/Vecgeom.test.cc [hunk 3]
```diff
@@ -1237,7 +1210,7 @@ TEST_F(FourLevelsGeantTest, tracking)
     }
     {
         SCOPED_TRACE("From outside edge");
-        auto result = this->track({-24, 10., 10.}, {1, 0, 0});
+        auto result = this->track({-24.1, 10., 10.}, {1, 0, 0});
```
Comment (sethrj) [inline test/geocel/vg/Vecgeom.test.cc]: This is no longer the outside edge as the test description shows. I assume you did this because it's failing with the surface geometry? It's important to keep this edge case—if we need to, we can skip this one for the surface geometry.

File: test/geocel/vg/Vecgeom.test.cc [hunk 4]
```diff
@@ -1278,7 +1251,7 @@ TEST_F(FourLevelsGeantTest, tracking)
     {
         // Formerly in linear propagator test, used to fail
         SCOPED_TRACE("From just outside world");
-        auto result = this->track({-24, 10, 10}, {1, 0, 0});
+        auto result = this->track({-24.1, 10, 10}, {1, 0, 0});
```
Comment (sethrj) [inline test/geocel/vg/Vecgeom.test.cc]: Looks like this test is actually a duplicate of the one above: I'm just going to delete this one.

**Review by sethrj — APPROVED** [review]

File: test/geocel/vg/Vecgeom.test.cc (hunk 3 above)
Comment (mrguilima) [reply test/geocel/vg/Vecgeom.test.cc]: The reason I moved the point was because the behavior is drastically different between v1.* (points on the border of the world are [OUTSIDE]) and v2.* (those points are *in* the World volume).
Hence all the arrays had a different number of elements.
I guess your suggestion is best handled in a test without propagation.

File: test/geocel/vg/Vecgeom.test.cc (hunk 4 above)
Comment (mrguilima) [reply test/geocel/vg/Vecgeom.test.cc]: The difference is the Geant vs. no-Geant, right?  Converted geometry vs. VGDML-based?
I thought you wanted to keep because of that.  I am fine with removing one of them.

Comment (sethrj) [reply test/geocel/vg/Vecgeom.test.cc]: I meant it's a duplicate of `From outside edge` a few lines above 😅

File: test/geocel/vg/Vecgeom.test.cc (hunk 3 above)
Comment (sethrj) [reply test/geocel/vg/Vecgeom.test.cc]: I get it, but a very important purpose for unit testing is to test and document edge cases and how they change from one version to another. What we *might* want to do is add code for the volume-based geometry so that, when initialized outside or on a surface, it moves to the nearest volume.
